src/Core__Promise.res

```diff
@@ -1,15 +1,15 @@
 type t<+'a> = promise<'a>
 
 @new
-external make: ((@uncurry (. 'a) => unit, (. 'e) => unit) => unit) => t<'a> = "Promise"
+external make: ((@uncurry 'a => unit, 'e => unit) => unit) => t<'a> = "Promise"
```

Comment:
`@uncurry` annotations can be removed?

Reply:
@glennsl I'll go ahead and just remove the `@uncurry` annotations, and we can let 1.0.0+ of Core be focused on uncurried mode (<1.0.0 can be used with curried mode in v11+ still). It'll be uncurried once it's in the compiler anyway, so seems like the best path forward.